Mathlib/Algebra/Polynomial/Splits.lean

```diff
@@ -139,6 +139,21 @@ theorem splits_X_pow (n : ℕ) : (X ^ n).Splits i :=
 theorem splits_id_iff_splits {f : K[X]} : (f.map i).Splits (RingHom.id L) ↔ f.Splits i := by
   rw [splits_map_iff, RingHom.id_comp]
 
+theorem Splits.comp_of_degree_one {f : K[X]} {p : K[X]} (hd : p.degree = 1)
+    (hl : IsUnit p.leadingCoeff) (h : f.Splits i) : (f.comp p).Splits i := by
```

Comment:
You should only need `IsUnit (p.map i).leadingCoeff` and this is automatic if `(p.map i).degree = 1`.

Reply:
I've done a little differently. I realized that I could further generalize the result to `(p.map i).degree ≤ 1`. Then prove a weaker but easier to check version for `p.degree ≤ 1`.

Currently, I write the conditions in the form `(p.map i).degree` instead of `(p.map i) = C a * X + C b`.

Reply:
Also added an iff version for `(p.map i).degree = 1`.